### ISSUE TITLE
Search path to link dependencies should search for vcpkg built products first

### ISSUE THREAD

**Issue by mheyman** [Microsoft/vcpkg#90]
The search path for libraries a build depends on should first include products that vcpkg built (under the _installed_ directory) rather than searching first along the environment's path.

For example, when doing `vcpkg install libpng`, which depends on zlib, if you have Anaconda installed, the build will find Anaconda's zlib rather than vcpkg's zlib and attempt a link with that (this fails).

**Comment by ras0219-msft** [comment]
I'm having a hard time reproducing this issue. Do you know what mechanism is picking up the alternate zlib?

From https://cmake.org/cmake/help/latest/command/find_library.html:

> 1. Search paths specified in cmake-specific cache variables. These are intended to be used on the command line with a -DVAR=value. This can be skipped if NO_CMAKE_PATH is passed.
>   * &#060;prefix>/lib/&#060;arch> if CMAKE_LIBRARY_ARCHITECTURE is set, and &#060;prefix>/lib for each &#060;prefix> in CMAKE_PREFIX_PATH
>   * CMAKE_LIBRARY_PATH
>   * CMAKE_FRAMEWORK_PATH

Since we add our paths to `CMAKE_PREFIX_PATH`, I'm not sure how Anaconda could be snaking ahead of us in the search order.

**Comment by mheyman** [comment]
On Thu, Sep 29, 2016 at 3:37 PM, Robert Schumacher
<notifications@github.com> wrote:
> I'm having a hard time reproducing this issue. Do you know what mechanism is
> picking up the alternate zlib?
> [SNIP]
I'm not sure - I'm more of a cmake hack then an expert.

In the build-x86-windows-rel-out.log file, the first time Anaconda3
shows up is the "Link:" log line. I removed Anaconda3 from my path and
the build worked.

I stole code from http://stackoverflow.com/a/9328525/240845 to dump
all the variables just before the find_library() call in
FindZLIB.cmake and the CMAKE_PREFIX_PATH is set correctly.

I don't know what is going on under the hood but I added
NO_SYSTEM_ENVIRONMENT_PATH to the find_path() and find_library() calls
FindZLIB.cmake and that also got my build to succeed.
